libqpdf/QPDFObjectHandle.cc

```diff
@@ -2156,9 +2192,25 @@ QPDFObjectHandle::parseInternal(PointerHolder<InputSource> input,
                     }
                     dict[key] = val;
                 }
+		if (!contents_string.empty() &&
+		    dict.count("/Type") &&
+		    dict["/Type"].isName() &&
+		    dict["/Type"].getName() == "/Sig" &&
+		    dict.count("/ByteRange") &&
+		    dict.count("/Contents") &&
+		    dict["/Contents"].isString())
+		{
+		    dict["/Contents"]
+		      = QPDFObjectHandle::newString(contents_string);
+		    dict["/Contents"].setParsedOffset(contents_offset);
+		}
                 object = newDictionary(dict);
                 setObjectDescriptionFromInput(
                     object, context, object_description, input, offset);
+                // The `offset` points to the next of "<<".
+                // Set the rewind offset to point to the beginning of "<<".
+                object.setParsedOffset(offset - 2);
```

Comment:
Same comment applies here as in the array case.

Reply:
Also `input->getLastOffset()` points the offset of ">>" that is closing dictionary instead of opening dictionary.
`qpdf/qtest/qpdf/minimal.pdf` used in the tests has several whitespace pattern before and after the "<<".